pkg/pods/manifest.go

```diff
@@ -28,31 +27,132 @@ type LaunchableStanza struct {
 	CgroupConfig            cgroups.Config `yaml:"cgroup,omitempty"`
 }
 
-type Manifest struct {
+type SignedManifest interface {
+	SignatureData() (plaintext, signature []byte)
+}
+
+type Manifest interface {
+	ID() string
+	RunAsUser() string
+	IsSigned() bool
+	Write(out io.Writer) error
+	ConfigFileName() (string, error)
+	WriteConfig(out io.Writer) error
+	PlatformConfigFileName() (string, error)
+	WritePlatformConfig(out io.Writer) error
+	GetLaunchableStanzas() map[string]LaunchableStanza
+	GetConfig() map[interface{}]interface{}
+	SHA() (string, error)
+	OriginalBytes() ([]byte, error)
+	SetID(string) error
+	SetConfig(config map[interface{}]interface{})
+	AddLaunchableStanza(launchableId string, stanza LaunchableStanza)
+	GetStatusPort() int
+	GetStatusHTTP() bool
+	SetRunAsUser(user string) error
+	SetStatusPort(port int)
```

Comment:
should we make `pods.Manifest` immutable and only `pods.UnsignedManifest` have setters?